jcapi-tags.go

```diff
@@ -42,84 +51,59 @@ func GetTagNames(tags []JCTag) []string {
 
-	if tag.Id != "" {
-		builder = append(builder, buildJSONKeyValuePair("_id", tag.Id))
-	}
+		tagResults := JCTagResults{}
 
-	builder = append(builder, buildJSONKeyValuePair("name", tag.Name))
-	builder = append(builder, buildJSONKeyValuePair("groupname", tag.GroupName))
-	builder = append(builder, buildJSONStringArray("systems", tag.Systems))
-	builder = append(builder, buildJSONStringArray("systemusers", tag.SystemUsers))
-	builder = append(builder, buildJSONStringArray("regularExpressions", tag.RegularExpressions))
-	builder = append(builder, buildJSONKeyValuePair("expirationTime", tag.ExpirationTime))
-	builder = append(builder, buildJSONKeyValueBoolPair("expired", tag.Expired))
-	builder = append(builder, buildJSONKeyValueBoolPair("selected", tag.Selected))
+		// Nope, must be a results array...
+		err = json.Unmarshal(result, &tagResults)
+		if err != nil {
+			err = fmt.Errorf("Could not unmarshal result '%s', err='%s'", string(result), err.Error())
+		}
 
-	builder = append(builder, buildJSONKeyValueBoolPair("externallyManaged", tag.ExternallyManaged))
-	builder = append(builder, buildJSONKeyValuePair("externalDN", tag.ExternalDN))
-	builder = append(builder, buildJSONKeyValuePair("externalSourceType", tag.ExternalSourceType))
+		tags = tagResults.Results
+	} else {
+		tags = append(tags, tag)
+	}
 
-	return []byte("{" + strings.Join(builder, ",") + "}"), nil
+	return
 }
 
-func (jc JCAPI) getTagFieldsFromInterface(tagData map[string]interface{}, tag *JCTag) {
-	tag.Id = tagData["_id"].(string)
-	tag.Name = tagData["name"].(string)
-
-	if tagData["groupName"] != nil {
-		tag.GroupName = getStringOrNil(tagData["groupName"].(interface{}))
-	}
+func (jc JCAPI) GetTagsByUrl(urlPath string) (tagList []JCTag, err JCError) {
 
-	if tagData["expirationTime"] != nil {
-		tag.ExpirationTime = getStringOrNil(tagData["expirationTime"].(interface{}))
-	}
-
-	if tagData["expired"] != nil {
-		tag.Expired = tagData["expired"].(bool)
+	result, err := jc.DoBytes(MapJCOpToHTTP(Read), urlPath, []byte{})
+	if err != nil {
+		return nil, fmt.Errorf("ERROR: Get tags from JumpCloud failed with urlPath='%s', err='%s'", urlPath, err.Error())
 	}
 
-	tag.Systems = jc.extractStringArray(tagData["systems"].([]interface{}))
-	tag.SystemUsers = jc.extractStringArray(tagData["systemusers"].([]interface{}))
-	tag.RegularExpressions = jc.extractStringArray(tagData["regularExpressions"].([]interface{}))
-
-	tag.ExternallyManaged = tagData["externallyManaged"].(bool)
-
-	if tagData["externalDN"] != nil {
-		tag.ExternalDN = getStringOrNil(tagData["externalDN"].(interface{}))
+	tagList, err = getJCTagsFromResults(result)
+	if err != nil {
+		return nil, fmt.Errorf("ERROR: Could not get tags from results, err='%s'", err.Error())
 	}
 
-	if tagData["externalSourceType"] != nil {
-		tag.ExternalSourceType = getStringOrNil(tagData["externalSourceType"].(interface{}))
-	}
+	return
 }
 
 func (jc JCAPI) GetAllTags() (tagList []JCTag, err JCError) {
+
 	var returnVal []JCTag
```

Comment:
The red/green is hard to read, but I think this isn't used in this function anymore?